File: old/start_fermenting.php
<?php
$stdin = fopen('php://stdin','r'); 

$recipes = json_decode(file_get_contents("recipes"),true);
foreach($recipes as $no=>$recipe); 
{
	$screen.=$recipe['name']." - ".$no."\r\n"; 
}
$screen.="Enter Recipe No#:\r\n"; 
echo $screen; 
$recipe_no = trim(fgets($stdin)); 

echo "Enter Fermenter Pos 1-4:\r\n"; 
$ferm_pos = trim(fgets($stdin));


$fermenting=json_decode(file_get_contents("fermenting"),true);
$fermenting[]=array("start_date"=>date("Y-m-d"),"recipe_no"=>$recipe_no,"ferm_pos"=>$ferm_pos);

file_put_contents("/opt/fermenters/fermenting",json_encode($fermenting)); 
?>

File: start_fermenting.py
#!/usr/bin/python
import json
from datetime import date
import sys
today = str(date.today())
screen =""
with open ('recipes.json') as data:
    recipes = json.load(data)
    for recipe_no in recipes:
      screen = screen + recipes[recipe_no]['name'] + ' - ' + recipe_no + '\r\n'
    screen = screen + 'Select Recipe #'
sel_recipe = input(screen)
ferm_pos = input('Enter Fermenter #: 1-4')

message = ""
with open ('fermenting.json') as data: 
    current_ferm = json.load(data)
    for item in current_ferm: 
        if item['ferm_pos'] == ferm_pos: 
            message = "There\'s already beer there."

if message != "": 
    print(message)
else: 
    fermenting = {"start_date":today,"recipe_no":sel_recipe,"ferm_pos":ferm_pos}
    with open('fermenting.json') as data:
        data = json.load(data)
        data.append(fermenting)
        with open('fermenting.json','w') as write:
            json.dump(data, write)

